server/src/main/java/org/elasticsearch/action/support/master/TransportMasterNodeAction.java

```diff
@@ -193,12 +183,41 @@ public void handleException(final TransportException exp) {
                         });
                     }
                 }
+            } catch (IndexNotFoundException e) {
```

Comment:
Not sure about catching an INFE thrown by any of this code, could we just protect the call to `checkBlock()`?